roofit/hs3/src/JSONFactories_RooFitCore.cxx

```diff
@@ -387,6 +448,58 @@ class RooHistFuncFactory : public RooJSONFactoryWSTool::Importer {
 };
 } // namespace
 
+
+#include <RooHistPdf.h>
+#include <TH1.h>
+
+namespace {
+class RooHistPdfStreamer : public RooJSONFactoryWSTool::Exporter {
+public:
+   std::string const &key() const override
+   {
+      static const std::string keystring = "histogramPdf";
+      return keystring;
+   }
+   bool exportObject(RooJSONFactoryWSTool *, const RooAbsArg *func, JSONNode &elem) const override
+   {
+      const RooHistPdf *hf = static_cast<const RooHistPdf *>(func);
+      const RooDataHist &dh = hf->dataHist();
+      elem["type"] << key();
+      RooArgList vars(*dh.get());
+      std::unique_ptr<TH1> hist{hf->createHistogram(RooJSONFactoryWSTool::concat(&vars).c_str())};
+      auto &data = elem["data"];
+      RooJSONFactoryWSTool::exportHistogram(*hist, data, RooJSONFactoryWSTool::names(&vars));
+      return true;
+   }
+};
+} // namespace
+
+namespace {
+class RooHistPdfFactory : public RooJSONFactoryWSTool::Importer {
+public:
+   bool importPdf(RooJSONFactoryWSTool *tool, const JSONNode &p) const override
+   {
+      std::string name(RooJSONFactoryWSTool::name(p));
+      if (!p.has_child("data")) {
```

Comment:
I only see a `data` child here. But to fully specify a RooHistPdf, don't you also need the bin boundaries? Where are they stored?

Reply:
The bin boundaries are stored in the observables.